From 4637f931f66566a9b3b65a4c45a54f5c0b77a0bf Mon Sep 17 00:00:00 2001
From: Martin Wilck <martin.wilck@ts.fujitsu.com>
Date: Wed, 24 Feb 2016 14:31:56 +0100
Subject: grub2: tboot doesn't skip first argument any more

Since change set 429:664e69, tboot doesn't skip over the first
command line argument any more. Remove code that inserted placeholders
to compensate the previous behavior.
---
 tboot/20_linux_tboot     |    8 ++++----
 tboot/20_linux_xen_tboot |   10 +++++-----
 2 files changed, 9 insertions(+), 9 deletions(-)

diff --git a/tboot/20_linux_tboot b/tboot/20_linux_tboot
index e8d2819..264098f 100644
--- a/tboot/20_linux_tboot
+++ b/tboot/20_linux_tboot
@@ -135,8 +135,8 @@ linux_entry ()
   printf '%s\n' "${prepare_boot_cache}"
   xmessage="$(gettext_printf "Loading tboot %s ..." ${tboot_version})"
   lmessage="$(gettext_printf "Loading Linux %s ..." ${version})"
-  tbargs=`merge_spaces ${rel_tboot_dirname}/${tboot_basename} ${tboot_args}`
-  lxargs=`merge_spaces ${rel_dirname}/${basename} root=${linux_root_device_thisversion} ro ${args} ${iommu_args} ${mb_extra_kernel}`
+  tbargs=`merge_spaces ${tboot_args}`
+  lxargs=`merge_spaces root=${linux_root_device_thisversion} ro ${args} ${iommu_args} ${mb_extra_kernel}`
   cat << EOF
 	echo	'$xmessage'
 	${mb_directive}	${rel_tboot_dirname}/${tboot_basename} ${tbargs}
@@ -147,7 +147,7 @@ EOF
     message="$(gettext_printf "Loading initial ramdisk ...")"
     cat << EOF
 	echo	'$message'
-	${mb_mod_directive} ${rel_dirname}/${initrd} ${rel_dirname}/${initrd}
+	${mb_mod_directive} ${rel_dirname}/${initrd}
 EOF
   fi
   if test -n "${sinit_list}" ; then
@@ -155,7 +155,7 @@ EOF
       message="$(gettext_printf "Loading sinit $i ...")"
       cat << EOF
 	echo	'$message'
-	${mb_mod_directive} ${rel_dirname}/${i} ${rel_dirname}/${i}
+	${mb_mod_directive} ${rel_dirname}/${i}
 EOF
     done
   fi
diff --git a/tboot/20_linux_xen_tboot b/tboot/20_linux_xen_tboot
index 5de2bb4..eeb16b6 100644
--- a/tboot/20_linux_xen_tboot
+++ b/tboot/20_linux_xen_tboot
@@ -126,9 +126,9 @@ linux_entry ()
   xmessage="$(gettext_printf "Loading Xen %s ..." ${xen_version})"
   tmessage="$(gettext_printf "Loading Tboot %s ..." ${tboot_version})"
   lmessage="$(gettext_printf "Loading Linux %s ..." ${version})"
-  tbargs=`merge_spaces ${rel_tboot_dirname}/${tboot_basename} ${tboot_args}`
-  xnargs=`merge_spaces placeholder ${xen_args} ${iommu_args}`
-  lxargs=`merge_spaces placeholder root=${linux_root_device_thisversion} ro ${args}`
+  tbargs=`merge_spaces ${tboot_args}`
+  xnargs=`merge_spaces ${xen_args} ${iommu_args}`
+  lxargs=`merge_spaces root=${linux_root_device_thisversion} ro ${args}`
   cat << EOF
 	echo	'$tmessage'
 	multiboot	${rel_tboot_dirname}/${tboot_basename} ${tbargs}
@@ -141,7 +141,7 @@ EOF
     message="$(gettext_printf "Loading initial ramdisk ...")"
     cat << EOF
 	echo	'$message'
-	module	${rel_dirname}/${initrd} ${rel_dirname}/${initrd}
+	module	${rel_dirname}/${initrd}
 EOF
   fi
   if test -n "${sinit_list}" ; then
@@ -149,7 +149,7 @@ EOF
       message="$(gettext_printf "Loading sinit $i ...")"
       cat << EOF
         echo    '$message'
-        module  ${rel_dirname}/${i} ${rel_dirname}/${i}
+        module  ${rel_dirname}/${i}
 EOF
     done
   fi
-- 
1.7.1

